### ISSUE TITLE
.ml stubs for exercises

### ISSUE THREAD

**Issue by sshine** [exercism/ocaml#285]
Some language tracks have a `{{ stub }}.ml` that the student can fill out.

The Ocaml track doesn't. Ideally, add stubs to all exercises.

To begin with, start with ones that use `Map` from Base, like the four exercises in #277.

Issue spawned from this message I wrote on Slack:

> I just reviewed the new version of Nucleotide Count and asked if there were any hard points, and *[the student]* said that the `(module Char)` annotations were difficult to figure out because they're not very well-documented. That was exactly the pain point that made it hard for me to convert the exercises from Core(_kernel) to Base. A solution could be to add a template .ml file with a stub in it that throws an exception. For some exercises, e.g. ones that require a data type definition repeated from the .mli, or in case of Nucleotide Count, a line like `let empty = Map.empty (module Char)`, that'd be really neat.

**Comment by stevejb71** [comment]
If it's too much effort, perhaps just prioritise the initial few exercises, and nucleotide count.

**Comment by sshine** [comment]
Another incident with a student spending way too much time (4 days) figuring out `Map.empty (module Char)` has pushed me to address this issue for Nucleotide Count. I'll look at the other exercises later.

**Comment by LucasMembrane** [comment]
I just spent a bunch of time on this with poor results, too.  One thing I did not get from the readme is that Map is in Base when Base is opened (which I think is true, now).  I was looking at the standard library, not aware that Base replaces Map.  The standard library base documentation works ok for a while on this problem,  but after a while the differences become more than apparent.      

Another odd thing is that when I googled for solutions,  the first three that I saw were apparently not doing the problem now out there -- I didn't see anything in them about the (Ok count) (Error c) (Ok map) returns, just counts and maps.  

I have so far found these exercises a fine way to learn,  so I have nothing but gratitude for your work, but since I came across this effort to improve things,  I figured I should share my POV.  Thanks much!
